### PR TITLE
Editing SC import function and plotting SSE

### PR DESCRIPTION
**Pull request by andrewram4287**
This repository is fixing the import function, making sure repetitive genes are removed, that the data is normalized via dividing by the sum of each gene, log transformed and the genes are being filtered. 

This repository is also calculating the SSE after performing NMF on the original dataset

### CODE REVIEW

File: gmm/figures/figure5.py [hunk 1]
```diff
@@ -23,19 +23,19 @@ def makeFigure():
...
-
-        maxloglikDF = pd.concat([maxloglikDF, row])
-
-    maxloglikDF = maxloglikDF.set_index("Rank")
-    sns.heatmap(data=maxloglikDF, ax=ax[0])
+    # maxloglikDF = pd.DataFrame(columns=["Rank", "Cluster", "MaxLoglik"])
+    # maxloglikDF = pd.DataFrame()
+    # for i in range(len(ranknumb)):
+    #     row = pd.DataFrame()
+    #     row["Rank"] = ["Rank:" + str(ranknumb[i])]
```
Comment (aarmey) [inline gmm/figures/figure5.py:30]: Do you mean for this to be commented out?

File: gmm/scImport.py [hunk 2]
```diff
@@ -17,7 +20,7 @@ def import_thompson_drug():
 
     metafile = pd.read_csv("gmm/data/meta.csv")  # Cell barcodes, sample id of treatment and sample number (33482,3)
     drugScreen = mmread("/opt/andrew/drugscreen.mtx").toarray()  # Sparse matrix of each cell/genes (32738,33482)-(Genes,Cell)
-    drugScreen = drugScreen.astype(np.float64)
+    drugScreen = drugScreen.astype(np.int32)
```
Comment (aarmey) [inline gmm/scImport.py:23]: Why an int?

File: gmm/scImport.py [hunk 3]
```diff
@@ -1,10 +1,13 @@
+import enum
+from inspect import getinnerframes
 import numpy as np
 import pandas as pd
 import csv
 import xarray as xa
 from sklearn.decomposition import NMF
 from scipy.io import mmread
 from scipy.stats import linregress
+import tensorly as tl
```
Comment (aarmey) [inline gmm/scImport.py:10]: I don't see tensorly used here?

File: gmm/scImport.py [hunk 4]
```diff
@@ -1,10 +1,13 @@
+import enum
+from inspect import getinnerframes
```
Comment (aarmey) [inline gmm/scImport.py:2]: Are these imports used?

File: gmm/scImport.py [hunk 5]
```diff
@@ -104,28 +109,34 @@ def geneNNMF(X, k=14, verbose=0, maxiteration=2000):
...
-def ThompsonDrugXA(numCells: int, rank: int, maxit: int):
+def ThompsonDrugXA(numCells: int, rank: int, maxit: int, r2x=False):
     """Converts DF to Xarray given number of cells, factor number, and max iter: Factor, CellNumb, Drug, Empty, Empty"""
     finalDF = pd.read_csv("/opt/andrew/FilteredDrugs_Offset1.3.csv")
+    # finalDF = pd.read_csv('FilteredDrugs_NoOffset.csv')
     finalDF.drop(columns=["Unnamed: 0"], axis=1, inplace=True)
     finalDF = finalDF.groupby(by="Drug").sample(n=numCells).reset_index(drop=True)
 
-    _, geneFactors = geneNNMF(finalDF, k=rank, verbose=0, maxiteration=maxit)
+    _, geneFactors, _ = geneNNMF(finalDF, k=rank, verbose=0, maxiteration=maxit)
```
Comment (aarmey) [inline gmm/scImport.py:133]: Since geneNNMF is already returning the SSE here, why not just always calculate R2X?

Comment (aarmey) [reply gmm/scImport.py:133]: Also, R2X is really only relevant for centered data, not NNMF. It's better to just report/plot SSE rather than R2X.

File: gmm/scImport.py [hunk 6]
```diff
@@ -143,4 +154,17 @@ def ThompsonDrugXA(numCells: int, rank: int, maxit: int):
             "Throwaway 1": ["Throwaway"],
             "Throwaway 2": ["Throwaway"],},)
 
-    return PopAlignXA
+    return PopAlignXA, rank_vector, varexpl_NMF
+
+def tensor_R2X(tensor, maxrank, maxit):
```
Comment (aarmey) [inline gmm/scImport.py:159]: Think you can delete this given the comment above.

**Review by aarmey — CHANGES_REQUESTED** [review]
Check the git action test. There's a testing error.

**Comment by andrewram4287** [comment]
> 

This test should be fixed now!

**Review by aarmey — APPROVED** [review]
Great. This all looks good to me. Let's just wait to see if it builds before merging.

**Comment by aarmey** [comment]
Great. Merging!
